community/kernel/src/main/java/org/neo4j/kernel/impl/newapi/Operations.java

```diff
@@ -563,11 +592,34 @@ private void acquireExclusiveNodeLock( long node )
         }
     }
 
+    private void acquireExclusiveRelationshipLock( long relationshipId )
+    {
+        if ( !ktx.hasTxStateWithChanges() || !ktx.txState().relationshipIsAddedInThisTx( relationshipId ) )
```

Comment:
I might be wrong here, but what about a transaction like this:

CREATE ()-[a:R]->()
CREATE  ()-[b:R]->()
DELETE a
DELETE b

Wouldn't b be unable to get the exclusive lock here?

Reply:
As both a and b are created in the transaction, no locks are needed or acquired for this case. The locks guard against concurrent modification of the nodes by other transactions, but as a and b are created in the current transaction, no other transaction will know about them until post commit, and therefor we do not need to lock them.